### ISSUE TITLE
Enum `@member` and `nonmember` mishandled in `Literal` types

### ISSUE THREAD

**Issue by sterliakov** [python/mypy#18719]
**Bug Report**

`typing.Literal` is supposed to accept enum members and not non-members (such as methods or explicit `nonmember` attributes). `mypy` fails to enforce the latter and support the former.

**To Reproduce**

```python
from enum import Enum, member, nonmember
from typing import Literal

class E(Enum):
    A = 1
    @member
    def B() -> None: ...
    C = member(lambda: None)
    D = nonmember(1)
    
    def meth(self) -> None: ...
    
a: Literal[E.A]  # OK, true negative
b: Literal[E.B]  # False positive: Parameter 1 of Literal[...] is invalid  [valid-type]
c: Literal[E.C]  # OK, true negative
d: Literal[E.D]  # False negative - `nonmember` shouldn't be allowed
meth: Literal[E.meth]  # OK, true positive: Parameter 1 of Literal[...] is invalid  [valid-type]
```

[playground](https://mypy-play.net/?mypy=latest&python=3.12&flags=strict&gist=826db3d3c8aeba4346fd4220c10097b3)

**Expected Behavior**

* The following should be accepted in Literal: `E.A`, `E.B`, `E.C` 
* The following should be rejected in Literal: `E.D`, `E.meth`

**Actual Behavior**

(line 7 is a different issue reported separately)

```
main.py:7: error: Method must have at least one argument. Did you forget the "self" argument?  [misc]
main.py:14: error: Parameter 1 of Literal[...] is invalid  [valid-type]
main.py:17: error: Parameter 1 of Literal[...] is invalid  [valid-type]
```

**Your Environment**

- Mypy version used: 1.15.0 and current master
- Mypy command-line flags: N/A
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.12
